src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,39 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+
+type TodoModalProps = {
+  selectedTodo: Todo | null;
+  closeModal: () => void;
+};
+
+export const TodoModal: React.FC<TodoModalProps> = (
+  { selectedTodo, closeModal },
+) => {
+  const [loading, setLoading] = useState(true);
+  const [user, setUser] = useState({ name: '', email: '' });
+
+  useEffect(() => {
+    if (selectedTodo) {
+      setLoading(true);
+      getUser(selectedTodo.userId)
+        .then(userData => {
+          setUser(userData);
+        })
+        .catch((error) => {
+          // eslint-disable-next-line no-console
+          console.error(error);
```

Comment:
remove comments and logs, and do something else on error, for ex. `setUser({ name: 'no user found' })`